message/rpc_protocol_impl.hh

```diff
@@ -180,24 +180,30 @@ auto send_message_cancellable(messaging_service* ms, messaging_verb verb, msg_ad
     auto rpc_client_ptr = ms->get_rpc_client(verb, id);
     auto& rpc_client = *rpc_client_ptr;
 
-    auto c = std::make_unique<seastar::rpc::cancellable>();
-    auto& c_ref = *c;
-    auto sub = as.subscribe([c = std::move(c)] () noexcept {
-        c->cancel();
+    struct state {
+        seastar::rpc::cancellable c;
+        std::exception_ptr ex;
+        optimized_optional<abort_source::subscription> sub;
+    };
+    auto st = make_lw_shared<state>();
+    auto& c_ref = st->c;
+    st->sub = as.subscribe([st] (const std::optional<std::exception_ptr>& opt_ex) noexcept {
+        st->c.cancel();
+        st->ex = opt_ex.value_or(std::make_exception_ptr(abort_requested_exception{}));
```

Comment:
A constructor in state would allow you to avoid optimized_optional. It's cosmetic, but one less thing to think about.